pytorch_lightning/callbacks/early_stopping.py

```diff
@@ -125,7 +125,7 @@ def __init__(
     def state_key(self) -> str:
         return self._generate_state_key(monitor=self.monitor, mode=self.mode)
 
-    def on_init_end(self, trainer: "pl.Trainer") -> None:
+    def setup(self, trainer: "pl.Trainer", pl_module: "pl.LightningModule", stage: Optional[str] = None) -> None:
```

Comment:
quick ques:
there is one more hook called after on_init_end and before setup, `on_before_accelerator_backend_setup`. Any reason why not use that one?

Reply:
no particular reason. I thought about it but thought `setup` is more generic and should work as well so went with setup. I can change it if you think it makes more sense

Reply:
Probably because setup() is more natural and the instructions inside this hook does not rely on whether accelerator is setup or not.

Reply:
either is fine. was just looking for a reason if there were any :)